plugins/hls-explicit-imports-plugin/src/Ide/Plugin/ExplicitImports.hs

```diff
@@ -67,37 +83,34 @@ descriptorForModules
       -- ^ Predicate to select modules that will be annotated
     -> PluginId
     -> PluginDescriptor IdeState
-descriptorForModules recorder pred plId =
+descriptorForModules recorder modFilter plId =
   (defaultPluginDescriptor plId)
     {
       -- This plugin provides a command handler
-      pluginCommands = [importLensCommand],
+      pluginCommands = [PluginCommand importCommandId "Explicit import command" (runImportCommand recorder)],
       -- This plugin defines a new rule
-      pluginRules = minimalImportsRule recorder,
-      pluginHandlers = mconcat
-        [ -- This plugin provides code lenses
-          mkPluginHandler SMethod_TextDocumentCodeLens $ lensProvider pred
+      pluginRules = minimalImportsRule recorder modFilter,
+      pluginHandlers =
+         -- This plugin provides code lenses
+           mkPluginHandler SMethod_TextDocumentCodeLens (lensProvider recorder)
+        <> mkPluginHandler SMethod_CodeLensResolve (lensResolveProvider recorder)
           -- This plugin provides code actions
-        , mkPluginHandler SMethod_TextDocumentCodeAction $ codeActionProvider pred
-        ]
-    }
-
--- | The command descriptor
-importLensCommand :: PluginCommand IdeState
-importLensCommand =
-  PluginCommand importCommandId "Explicit import command" runImportCommand
+        <> mkCodeActionHandlerWithResolve (codeActionProvider recorder) (codeActionResolveProvider recorder)
```

Comment:
Ah, so I think the thing I'm confused about is why we don't use `mkCodeActionHandlerWithResolveAndCommand` here. Then if we made `runImportCommand` handle everything then we could just use that logic for everything?

Reply:
Ah right, I see. Since we always need the handler for the lenses.